### PR TITLE
Check hyperwallet client response content type

### PR DESCRIPTION
**Pull request by akreisman-epam**

### CODE REVIEW

File: src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java [hunk 1]
```diff
@@ -132,6 +138,15 @@ protected void checkErrorResponse(final Response response) throws ParseException
         }
     }
 
+    private void checkResponseHeader(Response response) {
+        String contentTypeHeader = response.getHeader(CONTENT_TYPE_HEADER);
+        boolean isNotValidContentType = !contentTypeHeader.equals(VALID_JSON_CONTENT_TYPE) &&
+                !contentTypeHeader.equals(VALID_JSON_JOSE_CONTENT_TYPE);
```
Comment (peter-joseph) [inline src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java]: would this code be easier to read if we instead do the ff: `boolean isNotValidContentType = !(contentTypeHeader.equals(VALID_JSON_CONTENT_TYPE) || contentTypeHeader.equals(VALID_JSON_JOSE_CONTENT_TYPE))`

File: src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java [hunk 2]
```diff
@@ -132,6 +138,15 @@ protected void checkErrorResponse(final Response response) throws ParseException
...
+        if (isNotValidContentType) {
```
Comment (peter-joseph) [inline src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java]: can we also put the whole line 143 inside the if condition of line 145 the variable is not used elsewhere other than this if clause

File: src/test/java/com/hyperwallet/clientsdk/util/HyperwalletApiClientTest.java [hunk 3]
```diff
@@ -971,4 +1001,37 @@ public void testPost_500Response_WithRelatedResources() throws JSONException {
             assertThat(e.getResponse(), is(notNullValue()));
         }
     }
+
```
Comment (peter-joseph) [inline src/test/java/com/hyperwallet/clientsdk/util/HyperwalletApiClientTest.java]: Missing test where the test is passing with content-type response = `application/jose+json` ?

File: src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java (hunk 1 above)
Comment (akreisman-epam) [reply src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java]: Refactored this condition

File: src/test/java/com/hyperwallet/clientsdk/util/HyperwalletApiClientTest.java (hunk 3 above)
Comment (akreisman-epam) [reply src/test/java/com/hyperwallet/clientsdk/util/HyperwalletApiClientTest.java]: Added test

**Pull request merged**